test/common/http/http3/conn_pool_test.cc

```diff
@@ -0,0 +1,51 @@
+#include "common/http/http3/conn_pool.h"
+#include "common/quic/quic_transport_socket_factory.h"
+
+#include "test/common/upstream/utility.h"
+#include "test/mocks/common.h"
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/server/transport_socket_factory_context.h"
+#include "test/mocks/ssl/mocks.h"
+#include "test/mocks/upstream/cluster_info.h"
+#include "test/mocks/upstream/host.h"
+#include "test/test_common/simulated_time_system.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http3 {
+
+class Http3ConnPoolImplTest : public Event::TestUsingSimulatedTime, public testing::Test {
+public:
+  void initialize() {
+    EXPECT_CALL(*mock_host, address()).WillRepeatedly(Return(test_address_));
+    EXPECT_CALL(*mock_host, transportSocketFactory()).WillRepeatedly(testing::ReturnRef(factory_));
+    new Event::MockSchedulableCallback(&dispatcher_);
+    Network::ConnectionSocket::OptionsSharedPtr options;
+    Network::TransportSocketOptionsSharedPtr transport_options;
+    pool_ = allocateConnPool(dispatcher_, random_, host_, Upstream::ResourcePriority::Default,
+                             options, transport_options, state_, simTime());
+  }
+
+  NiceMock<Event::MockDispatcher> dispatcher_;
+  std::shared_ptr<Upstream::MockClusterInfo> cluster_{new NiceMock<Upstream::MockClusterInfo>()};
+  Upstream::MockHost* mock_host = new NiceMock<Upstream::MockHost>;
```

Comment:
Naming convention seems off for this field. Maybe just combine this and host_ into one field and avoid the raw pointer?